### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by musangamfure**
Hello Reviewer👋,

In this PR, I have completed all the requirements for the development and production parts of the Budget App capstone project.

**Here is what I have implemented:**

- [ ] Create the following pages:
    - Splash screen
    - Sign-up and login pages
    - Home page (categories page)
    - Transactions page
    - "Add a new category" page
    - "Add a new transaction" page
- [ ] Create unit and integration tests
- [ ] Technical requirements
    - Postgres for the database
    - Devise for authentication
    - Validate user input

- [ ] Deployed the app on render
- [ ] No linter error
- [ ] Gitflow follows

Kindly check my code and let me know what you think

Best Regards

Happy Coding🎉🎉

### CODE REVIEW

File: app/views/group_bills/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<div id="group_bills">
+  <% @group_bills.each do |group_bill| %>
+    <%= render group_bill %>
+    <p>
+      <%= link_to "Show this group bill", group_bill %>
+    </p>
+  <% end %>
+</div>
+
+<%= link_to "New group bill", new_group_bill_path %>
```
Comment (algerina) [inline app/views/group_bills/index.html.erb]: - [ ] You did a great job working on your app however you are required to match the given design as much as possible. Kindly  work on the highlighted section to match the design.


Your design:


Screenshot 2023-09-15 at 5 48 42 PM



The provided Design:

transactions_list (1)

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @musangamfure  ,

Great job so far! 

May I highlight the following: 
-  Your app is almost working well ✔
- The video presentation is descriptive✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

File: app/views/group_bills/index.html.erb (hunk 1 above)
Comment (musangamfure) [reply app/views/group_bills/index.html.erb]: Hello @algerina,

Thank you so much for your comment. But I would like to get clarification on what you want me to change. As you are pointing out in your screenshots, there are two pages. one is for individual groups, and the other is for all the transactions/groups. And I think they are all visible to  on my app.

here is my app:
groups

Comment (mckent05) [reply app/views/group_bills/index.html.erb:14]: - Hello @musangamfure, you have done well so far, but according to the previous reviewer, please style the transactions page as it is in the design template. From what the previous reviewer marked in the screenshot, the marked section, i.e the name of the category and the total amount, should be directly under the blue navbar. Please fix this

File: app/views/bills/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<p style="color: green"><%= notice %></p>
+<h1>Bills</h1>
+<div id="bills">
+  <% @bills.each do |bill| %>
+    <%= render bill %>
+    <p>
+      <%= link_to "Show this bill", group_bill_path(bill.group_id, bill)%>
+    </p>
+  <% end %>
+</div>
```
Comment (mckent05) [inline app/views/bills/index.html.erb]: - Hello @musangamfure, you have done well so far, but according to the previous reviewer, please style the transactions page as it is in the design template. From what the previous reviewer marked in the screenshot, the marked section, i.e the name of the category and the total amount, should be directly under the blue navbar. Please fix this

File: app/views/bills/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        <%= form.text_field :amount, class: 'form-control my-3', placeholder: 'Enter amount' %>
+        <%= form.hidden_field :group_id, {value: @group_id} %>
+      </div>
+      <div class="form-group">
+        <%= form.text_area :description, class: 'form-control my-3', placeholder: 'Add description' %>
+      </div>
+      <div class="form-group">
+        <%= form.submit 'Save Bill', class: 'btn btn-primary btn-sm bg-blue col-12' %>
+      </div>
+    <% end %>
```
Comment (mckent05) [inline app/views/bills/_form.html.erb]: **OPTIONAL** - According to the project requirements, when a user tries to add a new transaction, the following should be done:
`The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)`
Please make it possible to choose a category when adding a new transaction 😄

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @musangamfure, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost there


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍
- You have done well working on your capstone project 👍 👍 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
**OPTIONAL** - Please note that in your future capstone projects, when working on a solo capstone project your PR should merge from a feature branch into the dev branch. Please note this

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by abdulhamiid — APPROVED** [review]
Hi @musangamfure 👋, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**